docs/user_guide/scripting.rst

```diff
@@ -0,0 +1,78 @@
+Scripting
+==============================================
+
+
+Instead of configuring a project using a "clang-build.toml",
+you can also use a "clang-build.py" script.
+This allows you, for example, to generate a version-header
+based on git tags or to (de-)activate features or targets
+depending on environment variables.
+
+
+Script requirements
+----------------------------------------------
+
+The "clang-build.py" script, is required to define
+the following function:
+
+.. code-block:: Python
+
+    import clang_build
+
+    def get_project(directory, environment, parent=None) -> clang_build.project.Project:
+        project = clang_build.project.Project("projectname", {}, directory, environment, parent=parent)
+        #...
+        return project
+
+
+Creating a project
+----------------------------------------------
+
+You can default-initialize a project without targets
+or let clang-build create the project for you, from a
+folder or from a configuration `dict`.
+
+.. code-block:: Python
+
```

Comment:
I would put a `from clang_build.project import Project` to make all the following lines shorter.